FNX3-16641 ⁃ [Bug] Opening Fenix from sync notification causes Fenix to crash

## Steps to reproduce
1. Send a tab from another device
1. Wait 
1. Press notification to open received tab 

### Expected behavior
Fenix should open the tab and remove the notification. 

### Actual behavior
Fenix crashes (sometimes it opens the tab and doesn't remove the notification, but more often than not -- 95% of the time -- it crashes) 

### Device information
* Android device: Android 8, LG G6
* Fenix version: 1.0.1921

Crash log:
```
2019-05-27 12:10:13.818 30100-30100/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 30100
    java.lang.RuntimeException: Unable to resume activity {org.mozilla.fenix.debug/org.mozilla.fenix.HomeActivity}: java.lang.RuntimeException: Failed to call observer method
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3822)
        at android.app.ActivityThread.performNewIntents(ActivityThread.java:3172)
        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:3179)
        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:49)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:226)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:194)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:185)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:361)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:300)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:339)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:145)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:131)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2675)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:908)
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2087)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1861)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1817)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1717)
        at androidx.fragment.app.Fragment.performResume(Fragment.java:2684)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:915)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1228)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1293)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2646)
        at androidx.fragment.app.FragmentManagerImpl.dispatchResume(FragmentManagerImpl.java:2612)
        at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:268)
        at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:479)
        at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:468)
        at androidx.appcompat.app.AppCompatActivity.onPostResume(AppCompatActivity.java:195)
        at android.app.Activity.performResume(Activity.java:7325)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3814)
        at android.app.ActivityThread.performNewIntents(ActivityThread.java:3172) 
        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:3179) 
        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:49) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:2181)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:390)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:353)
2019-05-27 12:10:13.818 30100-30100/org.mozilla.fenix.debug E/AndroidRuntime:     at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:84)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:51)
        at mozilla.components.feature.session.EngineViewPresenter.start(EngineViewPresenter.kt:31)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:27)
        at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:157)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:193)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:216)
        	... 38 more
```

--- a/app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt
+++ b/app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt
@@ -54,6 +54,7 @@ class NotificationManager(private val context: Context) {
                 .setContentTitle(tab.title)
                 .setContentText(tab.url)
                 .setContentIntent(pendingIntent)
+                .setAutoCancel(true)
                 // Explicitly set a priority for <API25 devices.
                 // On newer devices this is inherited from the channel.
                 .setPriority(NotificationCompat.PRIORITY_HIGH)

